bindings/rust/s2n-tls-hyper/src/connector.rs

```diff
@@ -118,8 +118,21 @@ where
             return Box::pin(async move { Err(Error::InvalidScheme) });
         }
 
+        // IPv6 addresses are enclosed in square brackets within the host of a URI:
+        // https://datatracker.ietf.org/doc/html/rfc3986#section-3.2.2
+        //   IP-literal = "[" ( IPv6address / IPvFuture  ) "]"
+        //
+        // These square brackets aren't part of the domain itself, so they are trimmed off to ensure
+        // that the proper server name is provided to s2n-tls-tokio.
```

Comment:
Nit: Sometimes I have trouble reading big-brain RFC terminology. Could you include a concrete example in the comment showing `ipv6 from uri` and `ipv6 we need`.